### PR TITLE
Parser for indexer grpc

### PR DESCRIPTION
**Pull request by bowenyang007**
* WIP
- need to remove the fetcher and tailer: these are granted by grpc natively.
- change the code from `TransactionDataOnChain` to GRPC proto `transaction` as input.
- wire the individual processor.

### CODE REVIEW

**Comment by larry-aptos** [comment]
image

image


This is running now; some details needs to figure out

**Comment by larry-aptos** [comment]
Now the TPS is 15k. Rebase to the new thing(it's not necessary tho, i just realized..)

File: ecosystem/indexer-grpc/indexer-grpc-parser/README.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+### Miscellaneous
+1. If you run into
+   ```bash
+     = note: ld: library not found for -lpq
+             clang: error: linker command failed with exit code 1 (use -v to see invocation)
+   ```
+first make sure you have `postgres` and `libpq` installed via `homebrew`, see installation guide above for more details.
+This error is regarding the `libpq` library, a postgres C API library which diesel needs to run, more on this issue [here](https://github.com/diesel-rs/diesel/issues/2612).  
+
+2. [PostgreSQL Mac M1 installation guide](https://gist.github.com/phortuin/2fe698b6c741fd84357cec84219c6667).
```
Comment (saharct) [inline ecosystem/indexer-grpc/indexer-grpc-parser/README.md:111]: @bowenyang007 Does this also apply to M2 Macs?

File: ecosystem/indexer-grpc/indexer-grpc-parser/README.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+transactions in the fullnode with business logic from  each registered `TransactionProcessor`. On
+startup, by default, the indexer will restart from the first gap (e.g. version 5 if versions succeeded are 0, 1, 2, 3, 4, 6). 
+
+Each `TransactionProcessor` will need to be run in a separate fullnode. Please note that it may be difficult to run several transaction processors simultaneously in a single machine due to port conflicts. 
+
+When developing your own, ensure each `TransactionProcessor` is idempotent, and being called with the same input will not result in an error if some or all of the processing had previously been completed.
+
+## Requirements
+
+- [Rust](https://rustup.rs/)
```
Comment (saharct) [inline ecosystem/indexer-grpc/indexer-grpc-parser/README.md:21]: @bowenyang007 The installation for Rust needs to match the one provided in this [script](https://github.com/aptos-labs/aptos-core/blob/main/scripts/dev_setup.sh), right?

File: ecosystem/indexer-grpc/indexer-grpc-parser/README.md (hunk 1 above)
Comment (bowenyang007) [reply ecosystem/indexer-grpc/indexer-grpc-parser/README.md:111]: lol no idea. by the way this readme will be completely re-written. You might want to change the existing indexer readme instead.
https://github.com/aptos-labs/aptos-core/blob/aptos-grpc-indexer-client/crates/indexer/README.md

File: ecosystem/indexer-grpc/indexer-grpc-parser/README.md (hunk 2 above)
Comment (bowenyang007) [reply ecosystem/indexer-grpc/indexer-grpc-parser/README.md:21]: We'll have to revisit this as well lol. These instructions are just copied over from existing thing we'll clean up. Maybe it's good to just delete most of it and say that it's WIP.

File: ecosystem/indexer-grpc/indexer-grpc-parser/src/worker.rs [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+        };
+
+        let mut ma = MovingAverage::new(10_000);
+        info!(processor_name = processor_name, "[Parser] Starting stream");
+        loop {
+            let mut transactions_batches = vec![];
+            // Gets a batch of transactions from the stream. Batch size is set in the grpc server.
+            // The number of batches depends on our config
+            for _ in 0..concurrent_tasks {
+                let next_stream = match resp_stream.next().await {
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-parser/src/worker.rs:189]: try_next

File: ecosystem/indexer-grpc/indexer-grpc-parser/src/worker.rs [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+                            processor_name = processor_name,
+                            error = ?e,
+                            "[Parser] Error receiving datastream response"
+                        );
+                        break;
+                    },
+                    None => {
+                        // If no next stream wait a bit and try again
+                        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                        continue;
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-parser/src/worker.rs:203]: just break here.

File: ecosystem/indexer-grpc/indexer-grpc-parser/src/worker.rs [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+            processor_name = processor_name,
+            "[Parser] Running migrations"
+        );
+        self.run_migrations();
+        info!(
+            processor_name = processor_name,
+            "[Parser] Finished migrations"
+        );
+        // Connecting once to the RPC client.
+        // TODO: What happens if the connection is lost?
```
Comment (bowenyang007) [inline ecosystem/indexer-grpc/indexer-grpc-parser/src/worker.rs:115]: crashing is fine i guess.

**Pull request closed without merging**